lib/assets/javascripts/cartodb/models/filter.js

```diff
@@ -587,7 +591,8 @@ cdb.admin.models.Filters = Backbone.Collection.extend({
       return f.getSQLCondition();
     })
 
-    var sql = _(sqls).compact().join(' AND ');
+    var operator = this.dataLayer.get('sql_or_toggle') ? ' OR ' : ' AND ';
```

Comment:
Indeed, I thought about this and didn't bother because its a boolean toggle and the only operators that make sense here are AND/OR. However, its not more complicated to transform the boolean to a string operator at the view level so I think I'll do this.